### PR TITLE
Add e2e test to presubmit

### PR DESCRIPTION
**Pull request by johnugeorge**
Added grid example to the e2e test in presubmit

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/652)

### CODE REVIEW

File: test/e2e/v1alpha2/run-e2e-experiment.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	if exp.Status.Trials != *exp.Spec.MaxTrialCount {
+		log.Fatal("All trials are not run in the experiment ", exp.Status.Trials, exp.Spec.MaxTrialCount)
+	}
+
+	if exp.Status.TrialsSucceeded != *exp.Spec.MaxTrialCount {
+		log.Fatal("All trials are not successful ", exp.Status.TrialsSucceeded, *exp.Spec.MaxTrialCount)
+	}
+	if len(exp.Status.CurrentOptimalTrial.ParameterAssignments) == 0 {
```
Comment (richardsliu) [inline test/e2e/v1alpha2/run-e2e-experiment.go]: Should we check to see if exp.Status.CurrentOptimalTrial is null first?

File: test/e2e/v1alpha2/run-e2e-experiment.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		log.Fatal("Current Optimal Trial is empty")
+	}
+
+	log.Printf("Experiment has recorded best current Optimal Trial %v", exp.Status.CurrentOptimalTrial)
```
Comment (richardsliu) [inline test/e2e/v1alpha2/run-e2e-experiment.go]: Should also check the Observation and Metrics.

File: test/e2e/v1alpha2/run-e2e-experiment.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		log.Fatal("Yaml decode error ", err)
+	}
+	kclient, err := katibclient.NewClient(client.Options{})
+	if err != nil {
+		log.Fatal("NewClient for Katib failed: ", err)
+	}
+	var maxtrials int32 = 3
+	var paralleltrials int32 = 2
+	exp.Spec.MaxTrialCount = &maxtrials
+	exp.Spec.ParallelTrialCount = &paralleltrials
```
Comment (richardsliu) [inline test/e2e/v1alpha2/run-e2e-experiment.go]: We should set a metric here and make sure it is collected in the best trial.

File: test/e2e/v1alpha2/run-e2e-experiment.go (hunk 1 above)
Comment (johnugeorge) [reply test/e2e/v1alpha2/run-e2e-experiment.go]: done

File: test/e2e/v1alpha2/run-e2e-experiment.go (hunk 2 above)
Comment (johnugeorge) [reply test/e2e/v1alpha2/run-e2e-experiment.go]: done

File: test/e2e/v1alpha2/run-e2e-experiment.go (hunk 3 above)
Comment (johnugeorge) [reply test/e2e/v1alpha2/run-e2e-experiment.go]: done

**Pull request merged**